src/Illuminate/Routing/Matching/SubdomainValidator.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Illuminate\Routing\Matching;
+
+use Illuminate\Http\Request;
+use Illuminate\Routing\Route;
+
+class SubdomainValidator implements ValidatorInterface
+{
+    /**
+     * Validate a given rule against a route and request.
+     *
+     * @param  \Illuminate\Routing\Route  $route
+     * @param  \Illuminate\Http\Request  $request
+     * @return bool
+     */
+    public function matches(Route $route, Request $request)
+    {
+        if (is_null($route->subdomain())) {
+            return true;
+        }
+
+        $subdomains = implode('|', $route->subdomain());
+        $regex = "/^$subdomains\..*/i";
```

Comment:
Nope. This is why:

(PHP 5.6, 7 and 7.1)
```php
$subdomains = 'api|backend';
$regex = "/^$subdomains\..*/i";

var_dump(preg_match($regex, 'api.foo.co.uk')); // 1
var_dump(preg_match($regex, 'backend.foo.co.uk')); // 1
var_dump(preg_match($regex, 'foo.co.uk')); // 0

var_dump(preg_match($regex, 'api.localhost')); // 1
var_dump(preg_match($regex, 'backend.localhost')); // 1
var_dump(preg_match($regex, 'localhost')); // 0
```

The `\.` to avoid that (note the `api.`):
```php
// I prefer
Route::get('/posts', ['subdomain' => 'api', 'uses' => function() {}]);
// Than
Route::get('/posts', ['subdomain' => 'api.', 'uses' => function() {}]);
```

I think it's really simple but fully working, IMO we don't need to complexify the REGEX.